CMakeLists.txt

```diff
@@ -574,6 +566,16 @@ endif(MINGW OR MSYS)
 add_library(zlib SHARED ${ZLIB_SRCS} ${ZLIB_GZFILE_SRCS} ${ZLIB_ARCH_SRCS} ${ZLIB_ASMS} ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
 add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_GZFILE_SRCS} ${ZLIB_ARCH_SRCS} ${ZLIB_ASMS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
 
+if (ZLIB_COMPAT)
+    target_compile_definitions (zlib PUBLIC ZLIB_COMPAT)
+    target_compile_definitions (zlibstatic PUBLIC ZLIB_COMPAT)
+endif (ZLIB_COMPAT)
+
+if (WITH_GZFILEOP)
+    target_compile_definitions (zlib PUBLIC WITH_GZFILEOP)
+    target_compile_definitions (zlibstatic PUBLIC WITH_GZFILEOP)
```

Comment:
Or here... these are private definitions for internal use.

Reply:
But this defines used in zlib.h, without PUBLIC lib will be compiled with different contents of structure z_stream and functions like gzopen
Maybe this defines supposed to be in zconf.h ?

Reply:
@proller We already removed them from zlib.h, but that commit is not yet in "develop" branch.